### PR TITLE
Makefile: use normal conditionals

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR followups to  #14351 and cleanups some modified conditional that had been used as a workaround.

### Testing procedure

Same testing procedure as in #14351

### Related PR's

~~Depends on #14351~~

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
#14351 has been merged

**Comment by fjmolinas** [comment]
rebased

File: cpu/esp_common/Makefile.include [hunk 1]
```diff
@@ -62,9 +67,12 @@ ifneq (,$(filter lwip,$(USEMODULE)))
 endif
 
 # if SPI RAM is enabled, the qout flash mode has to be used
-# NOTE: These checks can be turned into normal conditionals when #9913 is fixed
-_FLASH_MODE_PREV := $(FLASH_MODE)
-FLASH_MODE = $(if $(filter esp_spi_ram,$(USEMODULE)),qout,$(_FLASH_MODE_PREV))
+  _FLASH_MODE_PREV := $(FLASH_MODE)
```
Comment (aabadie) [inline cpu/esp_common/Makefile.include:70]:
```suggestion
 _FLASH_MODE_PREV := $(FLASH_MODE)
```

File: boards/native/Makefile.include [hunk 2]
```diff
@@ -77,9 +77,10 @@ endif
 EEPROM_FILE ?= $(BINDIR)/native.eeprom
 
 # set the eeprom file flags only when the periph_eeprom feature is used.
-# NOTE: This can be turned into normal conditional syntax once #9913 is fixed
-EEPROM_FILE_FLAGS = $(if $(filter periph_eeprom,$(FEATURES_USED)),--eeprom $(EEPROM_FILE),)
-TERMFLAGS += $(EEPROM_FILE_FLAGS)
+ifneq (, $(filter periph_eeprom,$(FEATURES_USED)))
```
Comment (aabadie) [inline boards/native/Makefile.include:80]:
```suggestion
ifneq (,$(filter periph_eeprom,$(FEATURES_USED)))
```

Comment (fjmolinas) [reply boards/native/Makefile.include]: addressed

File: cpu/esp_common/Makefile.include (hunk 1 above)
Comment (fjmolinas) [reply cpu/esp_common/Makefile.include]: addressed

File: cpu/esp32/Makefile.include [hunk 3]
```diff
@@ -64,12 +66,18 @@ _ESP_WIFI_ANY_BASELIBS += -lphy -lstdc++
 # Baselibs needed when using esp_now module
 _ESP_NOW_BASELIBS = -lespnow -lmesh
 
-# NOTE: These checks can be turned into a normal conditional when #9913 is fixed
-BASELIBS += $(if $(filter esp_wifi_any,$(USEMODULE)),$(_ESP_WIFI_ANY_BASELIBS))
-BASELIBS += $(if $(filter esp_now,$(USEMODULE)),$(_ESP_NOW_BASELIBS))
+ifneq (,$(filter esp_wifi_any,$(USEMODULE)))
+  BASELIBS += $(_ESP_WIFI_ANY_BASELIBS)
```
Comment (gschorcht) [inline cpu/esp32/Makefile.include:70]: `_ESP_WIFI_ANY_BASELIBS` and `_ESP_NOW_BASELIBS` were only introduced for this work around and should be removed with this change.
```suggestion
  BASELIBS += -lcore -lrtc -lnet80211 -lpp -lcoexist -lwps -lwpa -lwpa2
  BASELIBS += -lphy -lstdc++
```

File: cpu/esp32/Makefile.include [hunk 4]
```diff
@@ -64,12 +66,18 @@ _ESP_WIFI_ANY_BASELIBS += -lphy -lstdc++
...
+endif
 
-UNDEF += $(if $(filter cpp,$(FEATURES_USED)),$(BINDIR)/esp_cxx/cxa_guard.o)
-BASELIBS += $(if $(filter cpp,$(FEATURES_USED)),-lstdc++)
+ifneq (,$(filter esp_now,$(USEMODULE)))
+  BASELIBS += $(_ESP_NOW_BASELIBS)
```
Comment (gschorcht) [inline cpu/esp32/Makefile.include:74]: See the comment above. Unfortunatly, multiline suggestion don't work for deleted lines.
```suggestion
  BASELIBS += -lespnow -lmesh
```

**Comment by fjmolinas** [comment]
@gschorcht addressed change requests, let me know if its OK.

**Comment by gschorcht** [comment]
@fjmolinas Please squash.

**Comment by fjmolinas** [comment]
I ran on 92d358bd5577b37cfd77635a68b4700c19a9c0f1

```
BUILD_IN_DOCKER=1 RIOT_MAKEFILES_GLOBAL_PRE= RIOT_MAKEFILES_GLOBAL_POST= time ./dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh build/deps/pr
```

and on 8c7b67702b02b69b757a80eb57da608952e31d38

```
BUILD_IN_DOCKER=1 RIOT_MAKEFILES_GLOBAL_PRE= RIOT_MAKEFILES_GLOBAL_POST= time ./dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh build/deps/master
```

```
diff build/deps/pr/deps_info build/deps/master/deps_info
# no diff
```

I have the output of both commands in a tarball, to big to post here directly, I can send it over to someone if you want to double check :)

**Comment by gschorcht** [comment]
> @aabadie All lights are green. Should we inform @miri64 before we merge due to soft feature freeze? I have requested changes to avoid accidental merging.

Sorry wrong PR :sunglasses:

**Comment by fjmolinas** [comment]
Thanks for the review!

**Pull request merged**